src/gui/transferlistfilterswidget.h

```diff
@@ -96,7 +100,8 @@ private slots:
     void updateTorrentStatus(const BitTorrent::Torrent *torrent);
     void updateTexts();
 
-    QMultiHash<const BitTorrent::Torrent *, TorrentFilter::Type> m_torrentsStatus;
+    using TorrentFilterBitset = std::bitset<sizeof(std::underlying_type_t<TorrentFilter::Type>) * 8>;
```

Comment:
It doesn't look like suitable. Their sizes have no such dependency.

Reply:
C++ doesn't have a way to get the number of elements in a `enum` and thus this workaround.

Reply:
You can have `enum` type with 256 different values with an internal type of only 1 byte in size. So that
`sizeof(std::underlying_type_t<Enum>) * 8` is 8 but you need 256.


> C++ doesn't have a way to get the number of elements in a `enum` and thus this workaround

Enumerations is one of the weaknesses of C++, IMO, even taking into account the improvements made in latest standards, and seeing the trends of its development, it will never be changed.

Reply:
I still think `QSet<TorrentFilter::Type>` would be the more convenient way.

Reply:
OK, changed to a (approximated) fixed value.

>I still think QSet<TorrentFilter::Type> would be the more convenient way.

While that is true, it is not preferable (to me) as it runs slower...

Reply:
> it runs slower...

In general, yes. But is there a noticeable difference in this case, when we have very few elements?

Reply:
>But is there a noticeable difference in this case, when we have very few elements?

I don't think it is noticeable (from users POV) however `StatusFilterWidget::updateTorrentStatus()` is invoked regularly on each UI update so I reckon it deserves performant code.